### ISSUE TITLE
Just Enough Items 1.12.2 4.15.0.276 requires Forge 2816 or higher

### ISSUE THREAD

**Issue by Titoo8899** [mezz/JustEnoughItems#1552]
Hi,
I love your mod and I love the functions. But I have a problem with your newest version. I play only with the 2768 Forge version. The higher versions of Forge work by the many of Mods what I have not. And I have with your mod no problems have. The last version 4.15.0.275 have with no problems work. But the 4.15.0.276 say in Minecraft this: 
You must include the right dependencies for Just Enough Items:
Requires Minecraft Forge 14.23.5.2816 or above (you have 14.23.5.2768)

You say: "Update to latest Forge in dev" in the Version: 4.15.0.276 for 1.12.2 Forge
And I have on Forge site als latest Forge version in dev that I have  thought this:
Download Recommended:
1.12.2 - 14.23.5.2768

Please fix this Problem why I can not update your mod then the Forge 14.23.5.2768 not more support is.

**Comment by Titoo8899** [comment]
The newest version of your mod 4.15.0.277 for 1.12.2 fix the problem with the Forge Version not. Please fix this why I can only with Forge 2768 play and not with 2816 or higher of the 1.12.2 Minecraft Forge version.

**Comment by mezz** [comment]
Thanks for the report!
Just Enough Items version 4.15.0.276 depends on a feature that only exists in Forge 14.23.5.2816 or above.
This is not a problem and I cannot remove the requirement.

To continue playing, you can either use a newer version of Forge or use an older version of JEI.

**Comment by MauveCloud** [comment]
Please at least explain what feature you're using that requires Forge build 2816.  I just checked, and as of writing this comment, 2768 is still the "recommended" (and stable) build of Forge for 1.12.2.  I doubt I'm the only player annoyed by mods requiring newer than recommended Forge (except for older MC versions like 1.7.10 where the "latest" is old enough that it's confusing that the "recommended" version is still different)

**Comment by mezz** [comment]
JEI uses the new `IRarity` interface introduced in Forge 2816.
It is listed in the [JEI changelog for 4.15.0.276](https://minecraft.curseforge.com/projects/jei/files/2698788):
```
Current release 4.15.0.276
Fix #1549 Add support for Forge IRarity for item tooltip colors
```

[Forge changelog for 14.23.5.2816 lists the change](https://files.minecraftforge.net/maven/net/minecraftforge/forge/1.12.2-14.23.5.2816/forge-1.12.2-14.23.5.2816-changelog.txt)

[The JEI commit is here](https://github.com/mezz/JustEnoughItems/commit/4dc46ad05c0a5ff67b31403f51bb9ad1cc94cf89)

In order to keep up with changes in Forge, mods need to develop for the latest version.
Sometimes this means adding support for new features that only exist in new Forge versions.
If you want to use the very latest JEI versions, you will occasionally need a version of Forge that is higher than the "recommended" version.

Speaking as a Forge developer, there isn't usually anything particularly unstable about the latest versions. If you want to use the latest mods, you don't need to be afraid of updating Forge to the version they require.

**Comment by bs2609** [comment]
> Please at least explain what feature you're using that requires Forge build 2816.

You can see the code added in the commit which increased the dependency version (4dc46ad05c0a5ff67b31403f51bb9ad1cc94cf89), as well as the referenced issue #1549.

Please also note that "recommended" != "stable", the recommended build 2768 was also the latest build when it was released. The decision to mark a build as recommended is the only formal requirement - no seperate extra testing is performed.

**Comment by MauveCloud** [comment]
> Please also note that "recommended" != "stable", the recommended build 2768 was also the latest build when it was released. The decision to mark a build as recommended is the only formal requirement - no seperate extra testing is performed.

That raises the question of what the point is of even having a "recommended" version, but to avoid (further) derailing the issue topic, perhaps one of you could clarify things in the related forum thread somebody else already posted: http://www.minecraftforge.net/forum/topic/65551-difference-between-recommended-and-latest-forge-version/

**Comment by smmmadden** [comment]
I understand the need to introduce new features, but couldn't the new feature be conditionalized on its own to check for build 2816+ and not cripple the mod for users on recommended builds?  Not every server owner will take dev builds until Forge marks them as recommended.  

Why add a feature that forces everyone to upgrade to a development build not yet recommended by Forge.  I too use this mod and love all the features, but sadly it means staying behind on hwyla updates waiting on Forge to catch up (again).

**Comment by mezz** [comment]
I am one of the Forge devs, I am “recommend”ing you update Forge past the pretty arbitrary recommended version if you want to use this version of JEI.
Either that or stop using the bleeding edge version of JEI, the bugs being fixed at this point in 1.12 tend to be super minor so you’re really not missing out on much by waiting a bit.

**Comment by juliusgaming14** [comment]
Did you try downloading the forge that Minecraft requires you to download? Because I have this problem before, then I followed the warning from Minecraft. Like I download everything that need to be required from mods, and somehow it worked. Try doing that!
